docs/internals/data-structures.rst

```diff
@@ -88,7 +88,10 @@ Segments
 ~~~~~~~~
 
 Objects referenced by a key are stored inline in files (`segments`) of approx.
-500 MB size in numbered subdirectories of ``repo/data``.
+500 MB size in numbered subdirectories of ``repo/data``. The number of segments
+per directory is controlled by the value of ``segments_per_dir``. If you change
+this value in a non-empty repository, you may also need to relocate the segment
+files manually.
```

Comment:
good update.

otoh, considering the default is now only 1000, I guess we are not likely to run into any problems with that (unlike with the 10000 it was before in 1.0).

so, we can add it or not, I am kind of indifferent about it.